### PR TITLE
Fix -f flag when downloading OS and always write version.txt when not explicitly setting a version

### PR DESCRIPTION
**Pull request by adrianstevens**

### CODE REVIEW

File: Source/v2/Meadow.Cli/Commands/Current/Firmware/FirmwareDownloadCommand.cs [hunk 1]
```diff
@@ -77,7 +75,7 @@ protected override async ValueTask ExecuteCommand()
             return;
         }
 
-        if (collection[Version] != null)
+        if (collection[Version] != null && Force == false)
```
Comment (stevenkuhn) [inline Source/v2/Meadow.Cli/Commands/Current/Firmware/FirmwareDownloadCommand.cs:78]: A small nit: Can this be `!Force`?

Comment (adrianstevens) [reply Source/v2/Meadow.Cli/Commands/Current/Firmware/FirmwareDownloadCommand.cs:78]: I personally prefer the long version ... I can parse it faster, the ! tends to blend into the variable name

File: Source/v2/Meadow.Cli/Commands/Current/Firmware/FirmwareDownloadCommand.cs [hunk 2]
```diff
@@ -61,6 +58,7 @@ protected override async ValueTask ExecuteCommand()
             }
 
             Logger?.LogInformation($"Latest available version is '{latest}'...");
+            await collection.SetDefaultPackage(latest);
```
Comment (stevenkuhn) [inline Source/v2/Meadow.Cli/Commands/Current/Firmware/FirmwareDownloadCommand.cs:61]: Is there going to be an issue setting the default package if there is an issue downloading that package?

Comment (ctacke) [reply Source/v2/Meadow.Cli/Commands/Current/Firmware/FirmwareDownloadCommand.cs:61]: Agreed.  I think the edge case bryan was having was "default is an older version.  I told it to download the latest.  I already had the latest, so it didn't set that as the default" where normally downloading latest sets it as the default.

Comment (stevenkuhn) [reply Source/v2/Meadow.Cli/Commands/Current/Firmware/FirmwareDownloadCommand.cs:61]: In the `if (collection[Version] != null && Force == false)` block starting on line 78, should the default package be set there? Effectively, if the latest version _is_ already available then go ahead and set the default package if it hasn't been updated already?

Comment (adrianstevens) [reply Source/v2/Meadow.Cli/Commands/Current/Firmware/FirmwareDownloadCommand.cs:61]: Yeah can do - depends on what you think the purpose of latest.txt is  ... should is record the latest version available or record the latest version on the dev system?

Either way we have to handle errors but if you have a strong pref I can change it.

**Review by stevenkuhn — APPROVED** [review]

File: Source/v2/Meadow.Cli/Commands/Current/Firmware/FirmwareDownloadCommand.cs (hunk 2 above)
Comment (adrianstevens) [reply Source/v2/Meadow.Cli/Commands/Current/Firmware/FirmwareDownloadCommand.cs]: ahh perfect - and fixed

**Review by ctacke — APPROVED** [review]

**Pull request merged**